src/controllers/signAccountOp/signAccountOp.ts

```diff
@@ -480,6 +480,12 @@ export class SignAccountOpController extends EventEmitter {
       this.estimation.erc4337GasLimits.gasPrice = bundlerGasPrices
     }
 
+    const shouldUpdateAccountOpCalls =
+      accountOp && accountOp.calls.length !== this.accountOp.calls.length
+    if (shouldUpdateAccountOpCalls) {
```

Comment:
The condition should check if (accountOp) as we may want to update other props of the object without affecting the number of calls

Reply:
Yes, makes sense. I did this in https://github.com/AmbireTech/ambire-common/pull/1011/commits/7ef3abd1a640557fbfe11e8cd3c328c911e3f091